lib/excon/connection.rb

```diff
@@ -252,6 +252,10 @@ def request(params={}, &block)
         datum[:headers]['Host']   ||= datum[:host] + port_string(datum)
       end
 
+      if host = datum[:headers].delete('Host')
```

Comment:
Overall this looks great, thanks for your detailed report and working through it with me. Just one small request and I think we should be good.

Could you add a comment here that notes the purpose (ensuring Host is first) and refers to the RFC # and section. I just worry that otherwise we may forget over time what the importance/intention of this is and it will be much easier to find/remember in a quick comment than to dig back through PRs and history. Does that seem reasonable to you?

Reply:
Hey! Sorry once again for the long delay.
Sure thing, I will soon add the comments as you requested.

Reply:
No problem, I appreciate it!